kannon/master.py

```diff
@@ -49,16 +49,17 @@ def build(self, root_task: gokart.TaskOnKart) -> None:
         task_queue = self._create_task_queue(root_task)
 
         # consume task queue
-        launched_task_ids: Set[str] = set()
         logger.info("Consuming task queue...")
         while task_queue:
             task = task_queue.popleft()
             if task.complete():
                 logger.info(f"Task {self._gen_task_info(task)} is already completed.")
                 continue
-            if task.make_unique_id() in launched_task_ids:
+            if task.make_unique_id() in self.task_id_to_job_name:
+                # check if task is still running on child job
+                assert self._check_child_task_status(task), f"Child task {self._gen_task_info(task)} failed."
```

Comment:
This assert will not be called, because this method raise error if job is not succeed.

Reply:
Thanks, fixed in d352a31.